src/sentry/integrations/discord/utils/auth.py

```diff
@@ -1,11 +1,13 @@
 from cryptography.exceptions import InvalidSignature
 from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PublicKey
+from rest_framework import status
 
 
-def verify_signature(public_key_string: str, signature: str, message: str) -> bool:
+def verify_signature(public_key_string: str, signature: str, timestamp: str, body: str) -> None:
     try:
         public_key = Ed25519PublicKey.from_public_bytes(bytes.fromhex(public_key_string))
-        public_key.verify(bytes.fromhex(signature), message.encode("utf-8"))
-        return True
+        public_key.verify(bytes.fromhex(signature), f"{timestamp}{body}".encode())
```

Comment:
Are we changing how we verify the signature? And do we not need the `if signature and timestamp` checks from `/discord/requests/base.py`?

Reply:
I'm following the Discord docs more explicitly https://discord.com/developers/docs/interactions/receiving-and-responding#security-and-authorization
If timestamp of body are None, body + timestamp can result in a TypeError. By placing the usage here, if anything fails in the try, it will be handled.

Reply:
Adding the if signature and timestamp checks back though